### PR TITLE
Rename S3 scrubber to storage scrubber

### PR DESCRIPTION
**Pull request by arpad-m**
The S3 scrubber contains "S3" in its name, but we want to make it generic in terms of which storage is used (#7547). Therefore, rename it to "storage scrubber", following the naming scheme of already existing components "storage broker" and "storage controller".

Part of #7547

### CODE REVIEW

**Review by VladLazar — COMMENTED** [review]
Looking at the code, the scrubber is S3 specific (it creates an s3 client). To my mind it would make more sense to make it generic first and rename it later. I don't mind too much though.

File: test_runner/regress/test_s3_scrubber.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from fixtures.common_types import TenantId, TenantShardId, TimelineId
 from fixtures.neon_fixtures import (
     NeonEnvBuilder,
-    S3Scrubber,
+    StorageScrubber,
```
Comment (VladLazar) [inline test_runner/regress/test_s3_scrubber.py:9]: nit: should this file be renamed to `test_storage_scrubber`?

File: storage_scrubber/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Neon S3 scrubber
+# Neon Storage Scrubber
 
 This tool directly accesses the S3 buckets used by the Neon `pageserver`
```
Comment (VladLazar) [inline storage_scrubber/README.md:3]: This still mentions S3, but it's correct since the scrubber doesn't support anything else atm :shrug:

**Review by VladLazar — APPROVED** [review]
I see no harm in merging this, but doing it now feels a bit awkward to me. In any case, feel free to merge.

File: storage_scrubber/README.md (hunk 2 above)
Comment (arpad-m) [reply storage_scrubber/README.md:3]: Yeah it's not the intended end state.

**Pull request merged**